platform/security/core/security-core-impl/src/main/java/ddf/security/samlp/impl/LogoutMessageImpl.java

```diff
@@ -112,16 +116,24 @@ public LogoutRequest extractSamlLogoutRequest(@NotNull String samlLogoutResponse
    *
    * @param nameIdString NameId of user to log out
    * @param issuerOrEntityId The Issuer of the LogoutRequest
+   * @param sessionIndexes
    * @return the built <code>LogoutRequest</code>
    */
   @Override
   public LogoutRequest buildLogoutRequest(
-      @NotNull String nameIdString, @NotNull String issuerOrEntityId) {
-    return buildLogoutRequest(nameIdString, issuerOrEntityId, UUID.randomUUID().toString());
+      @NotNull String nameIdString, @NotNull String issuerOrEntityId, List<String> sessionIndexes) {
+    return buildLogoutRequest(nameIdString, issuerOrEntityId, generateUuid(), sessionIndexes);
+  }
+
+  private String generateUuid() {
+    return getUuidGenerator().generateUuid();
   }
 
   public LogoutRequest buildLogoutRequest(
-      @NotNull String nameIdString, @NotNull String issuerOrEntityId, @NotNull String id) {
+      @NotNull String nameIdString,
+      @NotNull String issuerOrEntityId,
+      @NotNull String id,
+      @NotNull List<String> sessionIndexes) {
```

Comment:
❓ Do we need to check that `sessionIndexes` isn't `null` like we do for the other arguments?

Reply:
I don't know. I'll throw in a null check and see if it breaks anything. There are edge cases when the specification doesn't require a session, but I would rather have callers pass an empty list than a null value.